verilog/formatting/formatter_test.cc

```diff
@@ -5781,6 +5781,795 @@ static const std::initializer_list<FormatterTestCase> kFormatterTestCases = {
+        "    type TTT = C#(logic)\n"
+        ");\n"
+        "endinterface\n"
+    },
+    {   // parameterized interface with 'type list_of_type_assignments' and comments
+        "interface foo #(//comment\ntype T = int, type TT = bit, type TTT= C#(logic) );\n"
+        "endinterface\n",
+        "interface foo #(  //comment\n"
+        "    type T   = int,\n"
+        "    type TT  = bit,\n"
+        "    type TTT = C#(logic)\n"
+        ");\n"
+        "endinterface\n"
+    },
+    {   // parameterized interface with 'data_type list_of_param_assignments' and trailing comments
+        "interface foo #(type T = int, type TT = bit, //comment\n type TTT= C#(logic) );\n"
+        "endinterface\n",
+        "interface foo #(\n"
+        "    type T   = int,\n"
+        "    type TT  = bit,  //comment\n"
+        "    type TTT = C#(logic)\n"
+        ");\n"
+        "endinterface\n"
+    },
+    {   // parameterized interface with 'data_type list_of_param_assignments' and pre-proc
+        "interface foo #(type T = int,\n"
+        "`ifdef MACRO type TT = bit, `endif\n"
+        " type TTT= C#(logic));\n"
+        "endinterface\n",
+        "interface foo #(\n"
+        "    type T   = int,\n"
+        "`ifdef MACRO\n"
+        "    type TT  = bit,\n"
+        "`endif\n"
+        "    type TTT = C#(logic)\n"
+        ");\n"
+        "endinterface\n"
+    },
+    {   // parameterized interface with 'data_type list_of_param_assignments' and packed dimensions
+        "interface foo #(type T = int [3:0], type TT = bit [250:0]);\n"
+        "endinterface\n",
+        "interface foo #(\n"
+        "    type T  = int [  3:0],\n"
+        "    type TT = bit [250:0]\n"
+        ");\n"
+        "endinterface\n"
+    },
+    {   
+        "interface foo #(type T = int, "
+        "A = 2, int AA = 22, parameter AAA = 222, parameter longint AAAA = 2222, " 
+        "localparam AAAAA = 22222, localparam real AAAAAA = 222222"
+        ");\n"
+        "endinterface\n",
+        "interface foo #(\n"
+        "               type    T      = int,\n"
+        "                       A      = 2,\n"
+        "               int     AA     = 22,\n"
+        "    parameter          AAA    = 222,\n"
+        "    parameter  longint AAAA   = 2222,\n"
+        "    localparam         AAAAA  = 22222,\n"
+        "    localparam real    AAAAAA = 222222\n"
+        ");\n"
+        "endinterface\n"
+    },
+    {   // parameterized interface with built-in data type
+        "interface foo #(int a = 2, real abc = 2234);\n"
+        "endinterface\n",
+        "interface foo #(\n"
+        "    int  a   = 2,\n"
+        "    real abc = 2234\n"
+        ");\n"
+        "endinterface\n"
+    },
+    {   // parameterized interface with type
+        "interface foo #(type TYPE1 = int, type TYPE2 = boo);\n"
+        "endinterface\n",
+        "interface foo #(\n"
+        "    type TYPE1 = int,\n"
+        "    type TYPE2 = boo\n"
+        ");\n"
+        "endinterface\n"
+    },
```

Comment:
AWESOME set of test cases!

Few suggestions come to mind:

* test case(s) with `parameter type` together (or `localparam type`)
* one test case with a 2D array and 1D array (packed dimensions), since it looks like you've given them their own columns
* test cases with user defined types like `mytype_t` and `my_pkg::mytype_t` (since these have unqualified ids)

Reply:
Good suggestions.